### ISSUE TITLE
Too many small files written to object storage

### ISSUE THREAD

**Issue by Antaxify** [muhammadn/jaeger-s3#79]
Hi,

I gave this plugin a try, however I noticed it's writing lot's and lot's of small files (~900B each)to the object storage (GCS). Based on https://github.com/grafana/loki/issues/7851, I tried to increase `chunk_target_size`, `chunk_retain_period`, `chunk_idle_period`, but no luck so far. In our (quite large) setup, the plugin writes about 12 million files / hour.

Does the plugin write 1 chunk / span? Is it intended that it writes so many files or did I misconfigure it? Below is the config I use. Thanks for the help! :)

```
---
apiVersion: v1
data:
  plugin-config.yaml: |-
    schema_config:
      configs:
        - from: 2020-07-01
          store: boltdb-shipper
          object_store: gcs
          schema: v11
          index:
            prefix: index_
            period: 24h
          row_shards: 32

    storage_config:
      gcs:
        bucket_name: my-bucket-name
      boltdb_shipper:
        active_index_directory: /tmp/loki/boltdb-shipper-active
        cache_location: /tmp/loki/boltdb-shipper-cache
        cache_ttl: 24h         # Can be increased for faster performance over longer query periods, uses more disk space
        shared_store: gcs
      filesystem:
        directory: /tmp/loki/chunks

    compactor:
      working_directory: /tmp/loki/boltdb-shipper-compactor
      shared_store: gcs

    limits_config:
      max_streams_per_user: 0             # disable limit
      max_global_streams_per_user: 0      # disable limit
      enforce_metric_name: false
      reject_old_samples: true
      reject_old_samples_max_age: 168h

    memberlist:
      abort_if_cluster_join_fails: false
      max_join_backoff: 1m
      max_join_retries: 10
      min_join_backoff: 1s

    distributor:
      ring:
        kvstore:
          store: memberlist

    ingester:
      lifecycler:
        ring:
          kvstore:
            store: memberlist
          replication_factor: 1
        final_sleep: 0s
      chunk_idle_period: 10m
      chunk_retain_period: 60s
      chunk_target_size: 5572864
kind: ConfigMap
metadata:
  name: plugin-config
```
  
```
apiVersion: jaegertracing.io/v1
kind: Jaeger
metadata:
  name: jaeger-gcs
spec:
  serviceAccount: jaeger
  strategy: production
  collector:
    minReplicas: 6
    maxReplicas: 12
  storage:
    type: grpc-plugin
    grpcPlugin:
      image: ghcr.io/flitnetics/jaeger-objectstorage:latest
    options:
      grpc-storage-plugin:
        binary: /plugin/jaeger-objectstorage
        configuration-file: /plugin-config/plugin-config.yaml
        log-level: debug
  volumeMounts:
    - name: config-volume
      mountPath: /plugin-config
  volumes:
    - name: config-volume
      configMap:
        name: plugin-config
```

**Comment by muhammadn** [comment]
@Antaxify Hi, thanks for chiming in. I'll take a look into this. Have you done anything else since last week?

**Comment by diogoserrano** [comment]
Hello everyone,
I've tried this plugin, which apparently is working, to connect to a gcs, and i followed your yamls as an example @Antaxify thank you for that.
but i keep getting the same error:
HTTP Error: plugin error: rpc error: code = Unknown desc = Get "http://localhost:4100/loki/api/v1/query?query=count_over_time%28%7Benv%3D%22prod%22%7D%5B1h%5D%29&time=1684149247775031331&limit=1000": dial tcp 127.0.0.1:4100: connect: connection refused
Am i doing something wrong?

**Comment by tabramczyk** [comment]
Hello,

I share some observations bc maybe it can help you:
my apps **can't create enough logs to create good chunk before pod/node rotation**.
I.e. I've set _max_chunk_age_ and _chunk_idle_period_ to 12 hours but my app rotates after 2 hours (AutoScalling in Kubernetes).

I think good option could be compacting existing chunks and there is feature request here:
https://github.com/grafana/loki/issues/5605

**Comment by muhammadn** [comment]
@tabramczyk I'm working on compacting the existing chunks. I'll get back to you once i get this working.

@diogoserrano I think it is the deployment issue. Could you share your manifest if you are using kubernetes? (while omitting your sensitive information)

**Comment by muhammadn** [comment]
@Antaxify Sorry for the delay, I'm working on this as well right now. Will get back in this issue when i am done. Hopefully in a weeks time.

**Comment by muhammadn** [comment]
@Antaxify We will be releasing a new version soon but with a breaking change to how we store the traces. It should help with the small files issue. Data format will be different in this current version and we no longer will use loki to collect traces and send to s3.

We will update it here:

https://community.flitnetics.com/t/upcoming-update-of-jaeger-objectstorage/9

**Comment by muhammadn** [comment]
@Antaxify @tabramczyk  @diogoserrano 

I've released version v3.0.0 of this plugin. v3 uses tempo backend to store in GCS/AWS S3/AzureBlob instead of Loki since Tempo backend has matured so jaeger-objectstorage reads the traces from Tempo, so this would solve all your issues. 🎉

Closing for now. Do comment if you feel i'm missing something.
